### PR TITLE
Refactor DataFile API #553

### PR DESCRIPTION
**Pull request by LuciferYang**
## What changes were proposed in this pull request?
As #553 said:
1.DataFile: Change Signature of getFiberData and iterator method, remove configuration parameter
2.DataFile: remove getDictionary method from DataFile API, it  only use by OapDataFile.
3.OapIterator change to extends java.io.Closeable, java.lang.AutoCloseable allude we prefer to use "try with resouces " feature.

## How was this patch tested?
exiting test case.

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/io/ParquetDataFile.scala [hunk 1]
```diff
@@ -31,16 +30,16 @@ import org.apache.spark.sql.execution.datasources.parquet.ParquetReadSupportHelp
 import org.apache.spark.sql.types.StructType
 
 
-private[oap] case class ParquetDataFile
-(path: String, schema: StructType, configuration: Configuration) extends DataFile {
+private[oap] case class ParquetDataFile(path: String, schema: StructType,
+                                        configuration: Configuration) extends DataFile {
```
Comment (linhong-intel) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/io/ParquetDataFile.scala]: indent

**Comment by LuciferYang** [comment]
@linhong-intel ok~ already format ParquetDataFile.scala

**Comment by LuciferYang** [comment]
merge from upstream master & reslove conflict

**Comment by linhong-intel** [comment]
@LuciferYang conflict...

**Comment by LuciferYang** [comment]
@linhong-intel already fixed conflict :)

**Pull request merged**